staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,232 @@
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Items is a list of flow-schemas.
+	Items []FlowSchema `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// FlowSchemaSpec describes how the flow-schema's specification looks like.
+type FlowSchemaSpec struct {
+	// PriorityLevelConfigurationRef can reference a PriorityLevelConfiguration in the cluster. If the reference cannot
+	// be resolved, the flow-schema will ignored and marked as invalid in its status.
+	PriorityLevelConfiguration PriorityLevelConfigurationReference `json:"priorityLevelConfiguration,omitempty" protobuf:"bytes,1,opt,name=priorityLevelConfiguration"`
+	// MatchingPrecedence is used to choose among the FlowSchemas that match a given request.  The chosen
+	// FlowSchema is among those with the numerically lowest (which we take to be logically highest)
+	// MatchingPrecedence.  Each MatchingPrecedence value must be non-negative.
+	MatchingPrecedence int32 `json:"matchingPrecedence,omitempty" protobuf:"varint,2,opt,name=matchingPrecedence"`
+	// DistinguisherMethod defines how to classify requests into certain flows. If not empty, the flow's identifier is computed
+	// according to specification of the distinguisher.
+	// +optional
+	DistinguisherMethod FlowDistinguisherMethod `json:"distinguisherMethod,omitempty" protobuf:"bytes,3,opt,name=distinguisherMethod"`
+	// Rules holds all the RuleWithUser for this FlowSchema.
+	Rules []PolicyRuleWithSubject `json:"rules,omitempty" protobuf:"bytes,4,rep,name=rules"`
+}
+
+// FlowDistinguisherMethodType is the type of flow distinguisher method
+type FlowDistinguisherMethodType string
+
+// FlowDistinguisherMethod specifies the method of a flow distinguisher.
+type FlowDistinguisherMethod struct {
+	// Type is the type of flow distinguisher method
+	Type FlowDistinguisherMethodType `json:"type,omitempty" protobuf:"varint,1,opt,name=type"`
+}
+
+// PriorityLevelConfigurationReference contains information that points to the request-priority being used.
+type PriorityLevelConfigurationReference struct {
+	// Name is the name of resource being referenced
+	Name string `json:"name,omitempty" protobuf:"varint,1,opt,name=name"`
+}
+
+// PolicyRuleWithSubject prescribes a test that applies to a request to an apiserver. The test considers the subject
+// making the request, the verb being requested, and the resource to be acted upon.
+type PolicyRuleWithSubject struct {
+	// Subject is the normal user, serviceaccount, or group that this rule cares about.
+	// +optional
+	Subject rbacv1.Subject `json:"subject,omitempty" protobuf:"bytes,1,opt,name=subject"`
+	// Rule is the target verb, resource or the subresource the rule cares about. APIGroups, Resources, etc.
+	// Required.
+	Rule rbacv1.PolicyRule `json:"rule,omitempty" protobuf:"bytes,2,opt,name=rule"`
```

Comment:
It's a little dicey to reference this type, we'll have to fork it if e.g. rbac adds fields or we want different documentation. (not asking for a change, just a note)

Reply:
Yes.  But there is reason to expect continued alignment: we are trying to accomplish the same thing --- identify a set of <verb, resource>.  But it _is_ possible that RBAC might expand the mission of that type in the future.

Reply:
wait, isnt it a `v1` already?

Reply:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md explicitly contemplates changes to versioned types.  But I think rule 5 means that significant mission expansion can not happen.  So referencing the specific version of PolicyRule should be "safe".  But that still leaves open the question of what might we do in a future that has v2 of rbac.  But I think we should burn that bridge when we come to it.